debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/buffered/BufferedLogMinerStreamingChangeEventSource.java

```diff
@@ -1097,6 +1097,57 @@ private void logActiveTransactions() {
         }
     }
 
+    /**
+     * Abandons a single transaction identified by its transaction id.
+     * This method is public so it can be invoked by external signal actions.
+     *
+     * @param transactionId the transaction id to abandon, must be in lowercase hex format
+     * @return true if the transaction was found and abandoned, false otherwise
+     * @throws InterruptedException if interrupted while performing cleanup
+     */
+    public boolean abandonTransactionById(String transactionId) throws InterruptedException {
+        if (transactionId == null || transactionId.trim().isEmpty()) {
+            LOGGER.warn("Abandon transaction requested with null/empty transaction id");
+            return false;
+        }
+        final String txId = transactionId.trim();
```

Comment:
Since the signal handler trims the value, let's avoid a second call to `trim` and omit this.